.github/workflows/prod-deploy.yml

```diff
@@ -91,49 +95,85 @@ jobs:
+                fi
 
-            - name: Install/Update dependencies
+            - name: Run new container
               run: |
-                  cd ${{ env.DEPLOYMENT_PATH }}
-                  pip install --break-system-packages -r src/requirements.txt
+                docker run -d \
+                  --name ${{ env.CONTAINER_NAME }}-${{ env.IMAGE_TAG }} \
+                  --restart always \
+                  ghcr.io/${{ github.repository_owner }}/${{ env.IMAGE_NAME }}:${{ env.IMAGE_TAG }}
 
-            - name: Restart discord-bot with PM2
-              run: pm2 restart ${{ env.PM2_APP_NAME }}
 
             - name: Health Check
               run: |
-                  sleep 10 # Give bot time to start
-                  if pm2 describe ${{ env.PM2_APP_NAME }} | grep -q "online"; then
-                    echo "✅ Bot is running successfully"
-                  else
-                    echo "❌ Bot failed to start properly"
-                    pm2 logs ${{ env.PM2_APP_NAME }} --lines 10
-                    exit 1
-                  fi
+                sleep 10
+                if docker ps --filter "name=${{ env.CONTAINER_NAME }}-${{ env.IMAGE_TAG }}" --filter "status=running" --format '{{.Names}}' | grep -q "${{ env.CONTAINER_NAME }}-${{ env.IMAGE_TAG }}"; then
+                  echo "Production Container is running successfully"
+                else
+                  echo "Production Container failed to start"
+                  docker logs ${{ env.CONTAINER_NAME }}-${{ env.IMAGE_TAG }} --tail 20
+                  exit 1
+                fi
 
-    rollback:
-        name: Rollback on failure
-        runs-on: discord-bot-deploy-runner
-        needs: [merge, deploy]
-        if: failure() && needs.merge.result == 'success'
-        steps:
-            - name: Rollback to pre-merge commit
-              run: |
-                  cd ${{ env.DEPLOYMENT_PATH }}
-                  PRE_MERGE_COMMIT="${{ needs.merge.outputs.pre_merge_commit }}"
-                  echo "Rolling back to pre-merge commit: $PRE_MERGE_COMMIT"
-                  git reset --hard $PRE_MERGE_COMMIT
+    # rollback:
+    #     name: Rollback on failure
+    #     runs-on: discord-bot-deploy-runner
+    #     needs: [merge, deploy]
+    #     if: failure() && needs.merge.result == 'success'
+    #     steps:
+    #         - name: Rollback to pre-merge commit
+    #           run: |
+    #               cd ${{ env.DEPLOYMENT_PATH }}
+    #               PRE_MERGE_COMMIT="${{ needs.merge.outputs.pre_merge_commit }}"
+    #               echo "Rolling back to pre-merge commit: $PRE_MERGE_COMMIT"
+    #               git reset --hard $PRE_MERGE_COMMIT
 
-            - name: Restart with previous version
-              run: pm2 restart ${{ env.PM2_APP_NAME }}
+    #         - name: Restart with previous version
+    #           run: pm2 restart ${{ env.PM2_APP_NAME }}
```

Comment:
ultimately, i would need a rollback strategy. so think about how we can do that with the introduction of docker